[FEATURE] PAGEVIEW - Optimized integration of Page Rendering via Fluid (#1235)
TYPO3-Documentation/TYPO3CMS-Reference-Typoscript: Documentation/ContentObjects/Fluidtemplate/Index.rst

or to generate content within the top-level object page
(see :ref:`the example on this page <cobj-fluidtemplate-examples>`).

..  versionadded:: 13.1

    Starting with TYPO3 v 13.1 you can use the content object
    :ref:`PAGEVIEW <cobj-pageview>` for templates on page level. It reduces

>     the amount of TypoScript needed to render a page in the TYPO3 frontend.

Reviewer: Maybe also a sentence like:

"The new :typoscript:`PAGEVIEW` content object has very specific conventions and defaults, that requires (and allows) less configuration, with the benefit that following these conventions means less boilerplate code to maintain.
If you follow these conventions, a few directories and files most follow the structure outlined below."

?

Author: Added it, please have a look after your vacation.

.. _cobj-fluidtemplate-migration:

Migration from `FLUIDTEMPLATE` to `PAGEVIEW`
============================================

..  code-block:: typoscript

>     :caption: Before migration, EXT:my_sitepackage/Configuration/TypoScript/setup.typoscript

Author: I think it is a good idea. While we dont have a cs fixer for typoscript we can at least lint the files then and use the IDE support for formating

>     # Todo: migrate to PAGEVIEW

Author: This was to make clear that the following code is the old one.. but is probably confusing

>     # Todo: migrate to PAGEVIEW

Author: Removed it

            }

            layoutRootPaths {
                100 = EXT:my_sitepackage/Resources/Private/Layouts/Pages/
            }

            variables {
                pageUid = TEXT
                pageUid.data = page:uid

                pageTitle = TEXT
                pageTitle.data = page:title

                pageSubtitle = TEXT
                pageSubtitle.data = page:subtitle

                parentPageTitle = TEXT
                parentPageTitle.data = levelfield:-1:title
            }

            dataProcessing {
                10 = menu
                10.as = mainMenu
            }
        }
    }

.. code-block:: typoscript
    :caption: After migration, EXT:my_sitepackage/Configuration/TypoScript/setup.typoscript

    page = PAGE
    page {
        10 = PAGEVIEW
        10 {
            paths {
                100 = EXT:my_sitepackage/Resources/Private/Templates/
            }
            variables {
                parentPageTitle = TEXT
                parentPageTitle.data = levelfield:-1:title
            }
            dataProcessing {
                10 = menu
                10.as = mainMenu
            }
        }
    }

> In Fluid, the pageUid is available as :html:`{page.uid}` and pageTitle

Author: yep, forgot about that

as :html:`{page.title}`, the subtitle with :html:`{page.subtitle}`.

In this example some Fluid templates have to be moved. Move:

:path:`EXT:my_sitepackage/Resources/Private/Templates/Pages/`

>     The page Templates can stay in this folder.

Reviewer: ```suggestion
    The directory :file:`Templates` can stay in this structure.
```

(Is this still right then, did you mean this?)

Author: I dont think so, not talking about the directory

>     The page Templates can stay in this folder.

Author: ```suggestion
    The page-templates can stay in this folder.
```

:path:`EXT:my_sitepackage/Resources/Private/Partials/Pages/`
    Move files to :path:`EXT:my_sitepackage/Resources/Private/Templates/Partials/`
:path:`EXT:my_sitepackage/Resources/Private/Layouts/Pages/`

>     Move files to :path:`EXT:my_sitepackage/Resources/Private/Templates/Layouts/`

Reviewer: Could we maybe format those tasks into a bullet point list?

Author: Why not a definition list as we have now?

> If the private folder looked like this before:

Reviewer: ```suggestion
If the :file:`Private` folder looked like this before:
```

Example
=======

..  versionadded:: 13.1
    It is recommended to use :ref:`PAGEVIEW <cobj-pageview>` for page templates
    starting with TYPO3 v13.1. See
    :ref:`How to migrate to PAGEVIEW <cobj-fluidtemplate-migration>`

The Fluid template in

> :file:`EXT:my_sitepackage/Resources/Private/Templates/MyTemplate.html` could look

Reviewer 2: Maybe we can add a hint about the fact that the first letter has to be upper case in order to function well.

Author: This example here is still in the FLUIDTEMPLATE TypoScript object. I only added a note to send ppl over to the PAGEVIEW template

..  literalinclude:: _includes/_AfterMigrationToPageview.typoscript
    :language: typoscript
    :caption: After migration, EXT:my_sitepackage/Configuration/TypoScript/setup.typoscript

In Fluid, the pageUid is available as :fluid:`{page.uid}` and pageTitle
as :fluid:`{page.title}`, the subtitle with :fluid:`{page.subtitle}`.

In this example some Fluid templates have to be moved:

:path:`EXT:my_sitepackage/Resources/Private/Templates/Pages/`
    The page-templates can stay in this folder.
:path:`EXT:my_sitepackage/Resources/Private/Partials/Pages/`
    Move files to :path:`EXT:my_sitepackage/Resources/Private/Templates/Partials/`
:path:`EXT:my_sitepackage/Resources/Private/Layouts/Pages/`
    Move files to :path:`EXT:my_sitepackage/Resources/Private/Templates/Layouts/`

If the :path:`Private` folder looked like this before:

..  directory-tree::
    :level: 2

    *   :path:`EXT:my_sitepackage/Resources/Private/`

        *   :path:`Languages`
        *   :path:`Layouts`

            *   :path:`Pages`

        *   :path:`Partials`

            *   :path:`Pages`

        *   :path:`Templates`

>             *   :path:`Pages`

Author: I added a note

Example
=======

..  versionadded:: 13.1
    It is recommended to use :ref:`PAGEVIEW <cobj-pageview>` for page templates

>     starting with TYPO3 13.1. See

Reviewer: ```suggestion
    starting with TYPO3 v13.1. See
```
